src/statsmodels/formula.jl

```diff
@@ -234,34 +240,149 @@ function dropUnusedLevels!(da::PooledDataArray)
 end
 dropUnusedLevels!(x) = x
 
-function ModelFrame(trms::Terms, d::AbstractDataFrame)
+## whether or not a column of a particular type can be "promoted" to full rank
+## (only true of factors)
+can_promote(::PooledDataArray) = true
```

Comment:
Actually I now see this reflects are more fundamental property of the columns: whether they can give rise to several columns in the model matrix. I think it would be better to refer to this rather than to promotion (especially since we may need to export this to allow other types to override it). Maybe `iscategorical`?